lib/rspec/mocks/verifying_proxy.rb

```diff
@@ -193,6 +195,8 @@ def unimplemented?
       def self.for(object, method_name, proxy)
         if ClassNewMethodReference.applies_to?(method_name) { object }
           VerifyingExistingClassNewMethodDouble
+        elsif Mocks.configuration.temporarily_suppressing_partial_double_verification
+          MethodDouble
```

Comment:
It's not very clear to me why this is necessary.  Playing around with it locally, it looks like it's completely unnecessary for 2.x (at least, all the specs passed w/o this on 2.x!) but I can see that on 1.8.7, an argument failure occurs w/o this.

It'd be nice to do away with this, or, if we keep it only for 1.8.7, clearly document it so we know we can remove it when we remove 1.8.7.

That said, the fact that the arg verification caused the failure on 1.8.7 makes me realize that you are not specifically suppressing argument validation--which makes me worry that there could be some alternate way that it could be hit on other rubies.  So I think a better solution would be to override `validate_arguments!` on `VerifyingPartialDoubleProxy` to have it check the flag and then `super()` if needed.

Thoughts?